index.html

```diff
@@ -2540,18 +2605,438 @@ <h1>
+          </li>
+          <li>
+The <var>resolver options</var> are OPTIONAL for the <a>DID URL dereferencing</a> function.
+          </li>
+          <li>
+The `did` component of the <var>input <a>DID URL</a></var> is called the <var>input <a>DID</a></var>.
+          </li>
+          <li>
+The `method-name` component of the <var>input <a>DID</a></var> is called the <var>input <a>DID method</a></var>.
+          </li>
+        </ol>
+
+      </section>
+
+      <section>
+        <h3>
+Results
+        </h3>
+
+        <p>
+The results of the <a>DID URL dereferencing</a> function are:
+        </p>
+        <ol>
+          <li>
+A representation of the resource identified by the <var>input <a>DID URL</a></var>, e.g.:
+            <ul>
+              <li>
+A <a>DID document</a>, or
+              </li>
+              <li>
+A part of a <a>DID document</a>, or
+              </li>
+              <li>
+A <a>service endpoint</a> URL selected from a <a>DID document</a>, or
+              </li>
+              <li>
+Another resource as determined by the <var>input <a>DID method</a></var> and/or <a>DID controller</a>.
+              </li>
+            </ul>
+          </li>
+          <li>
+<a>DID document metadata</a> (see Section <a href="#did-document-metadata"></a>), if a <a>DID document</a> or a part of a <a>DID document</a> is returned. This component MAY be empty.
+          </li>
+          <li>
+<a>DID resolution metadata</a> (see Section <a href="#did-resolution-metadata"></a>).
+          </li>
+        </ol>
+
+        <p>
+The following table lists example <var>input <a>DID URLs</a></var> with the corresponding results
+of the <a>DID URL dereferencing</a> function:
+        </p>
+
+        <table class="example">
+          <thead>
+            <tr>
+              <th>
+Example Input DID URL
+              </th>
+              <th>
+Example Result
+              </th>
+            </tr>
+          </thead>
+
+          <tbody>
+            <tr>
+              <td>
+<code>did:example:1234</code>
+              </td>
+              <td>
+A DID document.
+              </td>
+            </tr>
+
+            <tr>
+              <td>
+<code>did:example:1234#key-1</code>
+              </td>
+              <td>
+A part of a DID document.
+              </td>
+            </tr>
+
+            <tr>
+              <td>
+<code>did:example:1234;service=agent</code>
+              </td>
+              <td>
+A service endpoint URL selected from a DID document.
+              </td>
+            </tr>
+
+            <tr>
+              <td>
+<code>did:example:1234/path</code><br>
+<code>did:example:1234?query=value</code>
+<code>did:example:1234/path?query=value</code><br>
+              </td>
+              <td>
+Another resource as determined by the input DID method and/or DID controller.
+              </td>
+            </tr>
+        </table>
+
+        <p>
+Implementation details and a concrete algorithm for this function can be found
```

Comment:
I get the feeling getting into the implementation details here are going to be the sticky part and it's best to leave this as implied or move it to an auxiliary implementation document which doesn't make normative statements. This probably means we'd have to pull the details of the algorithm out I would suspect.